keps/sig-node/2570-memory-qos/README.md

```diff
@@ -0,0 +1,121 @@
+
+cgroups v2 introduces a better way to limit and guarantee memory quality.
+
+| File | Description |
+| -------- | -------- |
+| memory.high | memory.high is the memory usage throttle limit. This is the main mechanism to control a cgroup’s memory use. If a cgroup's memory use goes over the high boundary specified here, the cgroup’s processes are throttled and put under heavy reclaim pressure. The default is max, meaning there is no limit. |
+| memory.max | memory.max is the memory usage hard limit, acting as the final protection mechanism: If a cgroup's memory usage reaches this limit and can't be reduced, the system OOM killer is invoked on the cgroup. Under certain circumstances, usage may go over the memory.high limit temporarily. When the high limit is used and monitored properly, memory.max serves mainly to provide the final safety net. The default is max. |
+| memory.low | memory.low is the best-effort memory protection, a “soft guarantee” that if the cgroup and all its descendants are below this threshold, the cgroup's memory won't be reclaimed unless memory can’t be reclaimed from any unprotected cgroups. |
+| memory.min | memory.min specifies a minimum amount of memory the cgroup must always retain, i.e., memory that can never be reclaimed by the system. If the cgroup’s memory usage reaches this low limit and can’t be increased, the system OOM killer will be invoked. |
+
+This proposal uses `memory.min` to retain request memory for `Guaranteed` pods and containers. And for `Burstable` pods, it uses `memory.low` to achieve best-effort memory protection at pod level.
+
+| qos class | pod level | container level | memory.min | memory.low |
+| -------- | -------- | -------- | -------- | -------- |
+| Guaranteed | y | y | requests | n |
+| Burstable | n | y | n | requests |
+
+### User Stories (Optional)
+- Workload is memory-sensitive, this feature can retain memory to reduce allocation latency
+- Workload is burstable, this feature can try best to protect request memory and decrease probability of memory being reclaimed
+
+### Comparison with Memory Manager
+The Memory Manager is a new component of kubelet ecosystem proposed to enable single-NUMA and multi-NUMA guaranteed memory allocation at topology level. And this Memory QoS proposal mainly uses the cgrups v2 mechanism to improve the quality of memory allocation, thereby improving the memory qos of guanranteed and burstable pods and even entire node. 
+See also https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1769-memory-manager
+
+### Risks and Mitigations
+- `Guaranteed` pod memory reservation could reduce node memory supply
+
+## Design Details
+### Feature Gate
+To set `--feature-gates=MemoryQOS=true` to enable the feature.
+
+### Rules
```

Comment:
maybe note here on the reasons behind these rules (as discussed in comments) of the reasoning of using `memory.min` vs `memory.low` for guaranteed vs burstable